### PR TITLE
[Fixed] exception when opening the virtual combo box without selection

### PR DESCRIPTION
**Pull request by mever**
A minor fix to prevent an exception from happening.

Cheers,

### CODE REVIEW

**Review by johnspackman — DISMISSED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by hkollmann — CHANGES_REQUESTED** [review]
Seems that some tests are broken...

**Review by hkollmann — CHANGES_REQUESTED** [review]
Could you try
 if (selection.length && (this.__convertValue(selection.getItem(0)) !== value))

?

**Review by derrell — DISMISSED** [review]

**Review by cboulanger — DISMISSED** [review]

**Review by oetiker — CHANGES_REQUESTED** [review]
this seems to break tests
https://travis-ci.org/qooxdoo/qooxdoo/jobs/396399588#L1075

**Comment by level420** [comment]
failing tests are:
```
Firefox 60.0.0 (Linux 0.0.0) qx.test.ui.form.virtual VirtualComboBox:testSelectFirstMatch FAILED
	Expected 'item 4' but found 'null'!
		qx.core.AssertionError.constructor:326
		qx.core.Assert.__fail:1179
		qx.core.Assert.assertEquals:4808
		qx.type.BaseString.assertEquals:779
		qx.test.ui.form.virtual.VirtualComboBox.testSelectFirstMatch:7
		qx.dev.unit.TestFunction:1438
		qx.dev.unit.TestResult.run:5892
		qx.dev.unit.TestFunction.run:1287
		.runTest://travis-job-qooxdoo-qooxdoo-396399588.travisci.net:9876node_modules/karma-qooxdoo/qooxdoo-adapter.js
Firefox 60.0.0 (Linux 0.0.0) qx.test.ui.form.virtual VirtualComboBox:testSelectFirstMatchWithSortedModel FAILED
	Expected 'item 49' but found 'null'!
		qx.core.AssertionError.constructor:326
		qx.core.Assert.__fail:1179
		qx.core.Assert.assertEquals:4808
		qx.type.BaseString.assertEquals:779
		qx.test.ui.form.virtual.VirtualComboBox.testSelectFirstMatchWithSortedModel:7
		qx.dev.unit.TestFunction:1438
		qx.dev.unit.TestResult.run:5892
		qx.dev.unit.TestFunction.run:1287
		.runTest://travis-job-qooxdoo-qooxdoo-396399588.travisci.net:9876node_modules/karma-qooxdoo/qooxdoo-adapter.js
.
Firefox 60.0.0 (Linux 0.0.0) qx.test.ui.form.virtual VirtualComboBox:testSelectFirstMatchWithFormatter FAILED
	Expected '<b>item 4</b>' but found 'null'!
		qx.core.AssertionError.constructor:326
		qx.core.Assert.__fail:1179
		qx.core.Assert.assertEquals:4808
		qx.type.BaseString.assertEquals:779
		qx.test.ui.form.virtual.VirtualComboBox.testSelectFirstMatchWithFormatter:7
		qx.dev.unit.TestFunction:1438
		qx.dev.unit.TestResult.run:5892
		qx.dev.unit.TestFunction.run:1287
		.runTest://travis-job-qooxdoo-qooxdoo-396399588.travisci.net:9876node_modules/karma-qooxdoo/qooxdoo-adapter.js
Firefox 60.0.0 (Linux 0.0.0) qx.test.ui.form.virtual VirtualComboBox:testSelectFirstMatchByLabelPath FAILED
	this.__comboBox.getChildControl(...)._preselected is null
		qx.test.ui.form.virtual.VirtualComboBox:25
		qx.dev.unit.TestFunction:1438
		qx.dev.unit.TestResult:5892
		qx.dev.unit.TestFunction:1287
```

**Comment by level420** [comment]
The problem is, that `__selectFirstMatch` also needs to execute its code event without an existing selection, as it has to run `dropdown.setPreselected(modelItem)` on the dropdown widget.

To solve this we have do ask for the selection not being empty OR the current value not being the current selection, only resetting the selection if there was one.

@mever I could hijack your PR and change it if you allow me to do so.

File: framework/source/class/qx/ui/form/VirtualComboBox.js [hunk 1]
```diff
@@ -359,7 +359,7 @@ qx.Class.define("qx.ui.form.VirtualComboBox",
       var dropdown = this.getChildControl("dropdown");
       var selection = dropdown.getSelection();
 
-      if (this.__convertValue(selection.getItem(0)) !== value)
+      if (selection.length && this.__convertValue(selection.getItem(0)) !== value)
```
Comment (level420) [inline framework/source/class/qx/ui/form/VirtualComboBox.js]: The problem is, that `__selectFirstMatch` also needs to execute its code even without an existing selection, as it has to run `dropdown.setPreselected(modelItem)` on the dropdown widget.

To solve this we have to ask for the selection not being empty OR the current value not being the current selection, only resetting the selection if there was one.

@mever I could hijack your PR and change it if you allow me to do so.

**Review by level420 — CHANGES_REQUESTED** [review]

File: framework/source/class/qx/ui/form/VirtualComboBox.js (hunk 1 above)
Comment (mever) [reply framework/source/class/qx/ui/form/VirtualComboBox.js]: Hi, sorry for the delay.
Be my guest :+1:

**Comment by level420** [comment]
@mever could you please check if my modification works for you? thank you.

**Review by hkollmann — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Comment by mever** [comment]
@level420 Tonight I have some time to check if your modification works for me too. But I do not mind if you merge it already.

**Comment by level420** [comment]
@mever did you find some time to verify this PR? It would be great to have your go for this. Thank you!

**Comment by mever** [comment]
@level420 , sorry for the delay. I did not explicitly tested the converter but the logic seems sound. It seems to work on our application. I can not find any problems.

**Comment by level420** [comment]
thank you @mever

**Review by level420 — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by cajus — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**